qdb/comm.py

```diff
@@ -658,36 +666,35 @@ def get_events_from_socket(sck):
     """
     while True:
         try:
-            rlen = sck.recv(4)
-            if len(rlen) != 4:
-                return
-            rlen = unpack('>i', rlen)[0]
-            bytes_received = 0
-            resp = b''
-            with Timeout(1, False):
-                while bytes_received < rlen:
-                    resp += sck.recv(rlen - bytes_received)
-                    bytes_received = len(resp)
+            sck.setblocking(True)
+            resp = bytearray(4)
+            if sck.recv_into(resp, 4) != 4:
```

Comment:
This should be a blocking call with no timeout. The loop is broken into two steps:

1. read the length
1. read n bytes representing the message.

The other side of this socket is not streaming messages constantly so we need to wait on a new message to come in. It will always send the length and the message together so once we have recieved the length, we should timeout on reading the body because that would signal that something is wrong.